tests/wallet/dao_wallet/test_dao_clvm.py

```diff
@@ -0,0 +1,973 @@
+    assert len(conds.as_python()) == 7 + len(conditions)
+
+    # test update
+    proposal: Program = proposal_curry_one.curry(
+        proposal_curry_one.get_tree_hash(),
+        proposal_id,
+        acs_ph,
+        950,
+        1200,
+    )
+    full_proposal = SINGLETON_MOD.curry(proposal_struct, proposal)
+    proposal_coin_id = Coin(parent_id, full_proposal.get_tree_hash(), proposal_amt).name()
+    solution = Program.to(
+        [
+            1000,
+            5100,
+            [proposal_coin_id, acs_ph, 0],
+            [proposal_id, 1200, 950, parent_id, proposal_amt],
+            [[51, 0xCAFEF00D, spend_amount]],
+        ]
+    )
+    conds: Program = proposal_validator.run(solution)
+    assert len(conds.as_python()) == 3
+
+    return
+
+
+def test_merge_p2_singleton() -> None:
+    # Setup a singleton struct
+    singleton_inner: Program = Program.to(1)
+    singleton_id: Program = Program.to("singleton_id").get_tree_hash()
+    singleton_struct: Program = Program.to((SINGLETON_MOD_HASH, (singleton_id, SINGLETON_LAUNCHER_HASH)))
+
+    # Setup p2_singleton_via_delegated puz
+    my_id = Program.to("my_id").get_tree_hash()
+    p2_singleton = P2_SINGLETON_MOD.curry(singleton_struct, P2_SINGLETON_AGGREGATOR_MOD)
+    my_puzhash = p2_singleton.get_tree_hash()
+
+    # Spend to delegated puz
+    delegated_puz = Program.to(1)
+    delegated_sol = Program.to([[51, 0xCAFEF00D, 300]])
+    solution = Program.to([0, singleton_inner.get_tree_hash(), delegated_puz, delegated_sol, my_id])
+    conds = conditions_dict_for_solution(p2_singleton, solution, INFINITE_COST)
+    apa = std_hash(
+        SINGLETON_MOD.curry(singleton_struct, singleton_inner).get_tree_hash()
+        + Program.to([my_id, delegated_puz.get_tree_hash()]).get_tree_hash()
+    )
+    assert len(conds) == 4
+    assert conds[ConditionOpcode.ASSERT_PUZZLE_ANNOUNCEMENT][0].vars[0] == apa
+    assert conds[ConditionOpcode.CREATE_COIN][0].vars[1] == int_to_bytes(300)
+
+    # Merge Spend (not output creator)
+    merge_sol = Program.to([[my_id, my_puzhash, 300, 0]])
+    conds = conditions_dict_for_solution(p2_singleton, merge_sol, INFINITE_COST)
+    assert len(conds) == 2
+    assert conds[ConditionOpcode.ASSERT_MY_PUZZLEHASH][0].vars[0] == my_puzhash
+    assert conds[ConditionOpcode.CREATE_COIN_ANNOUNCEMENT][0].vars[0] == int_to_bytes(0)
+
+    # Merge Spend (output creator)
+    fake_parent_id = Program.to("fake_parent").get_tree_hash()
+    merged_coin_id = Coin(fake_parent_id, my_puzhash, 200).name()
+    merge_sol = Program.to([[my_id, my_puzhash, 100, [[fake_parent_id, my_puzhash, 200]]]])
+    conds = conditions_dict_for_solution(p2_singleton, merge_sol, INFINITE_COST)
+    assert len(conds) == 5
+    assert conds[ConditionOpcode.ASSERT_COIN_ANNOUNCEMENT][0].vars[0] == Program.to([merged_coin_id, 0]).get_tree_hash()
+    assert conds[ConditionOpcode.CREATE_COIN][0].vars[1] == int_to_bytes(300)
+    return
+
+
+def test_treasury() -> None:
```

Comment:
hard to see which scenarios we're testing here, can you describe here please or create separate tests for each